src/plugins/kibana_react/public/page_template/no_data_page/no_data_page_body/no_data_page_body.tsx

```diff
@@ -0,0 +1,57 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import { EuiLink, EuiSpacer, EuiText, EuiTextColor } from '@elastic/eui';
+import React, { ReactElement } from 'react';
+import { FormattedMessage } from '@kbn/i18n-react';
+import { NoDataPageProps } from '../no_data_page';
+import { KibanaPageTemplateSolutionNavAvatar } from '../../solution_nav';
+import { ActionCard, ActionCards } from '../actions';
+
+type NoDataPageBodyProps = {
+  actionCards: Array<ReactElement<typeof ActionCard>>;
+} & Omit<NoDataPageProps, 'actions'>;
+
+export const NoDataPageBody = (props: NoDataPageBodyProps) => {
+  const { pageTitle, docsLink, solution, actionCards, logo } = props;
+
+  return (
+    <React.Fragment>
+      <EuiText textAlign="center">
+        <KibanaPageTemplateSolutionNavAvatar
+          name={solution}
+          iconType={logo || `logo${solution}`}
+          size="xxl"
+        />
+        <h1>{pageTitle}</h1>
+        <EuiTextColor color="subdued">
+          <p>
+            <FormattedMessage
+              id="kibana-react.noDataPage.intro"
+              defaultMessage="Add your data to get started, or {link} about {solution}."
+              values={{
+                solution,
+                link: (
+                  <EuiLink href={docsLink}>
+                    <FormattedMessage
+                      id="kibana-react.noDataPage.intro.link"
+                      defaultMessage="learn more"
+                    />
+                  </EuiLink>
+                ),
+              }}
+            />
+          </p>
+        </EuiTextColor>
+      </EuiText>
+      <EuiSpacer size="xxl" />
+      <EuiSpacer size="l" />
+      <ActionCards actionCards={actionCards} />
```

Comment:
I'm confused about this level of abstraction/refactor. All that these new components are are EuiFlexGrid/EuiFlexItems. They're simply layout utilities that don't contain anything more helpful than that. Can you explain what the thought was for the future of this?

Reply:
The idea was to have components that can be tested separately, outside of `PageTemplate` component

Reply:
Right, right, I understand the basic concept of why we'd want to breakdown to more sizable chunks, but I'm more questioning this one in particular. There's nothing more that `ActionCards` and `ActionCard` do than simply wrap `EuiFlexGrid` and `EuiFlexItem` respectively. These components purely serve as layout mechansisms and honestly a `EuiFlexItem` without a wrapping `EuiFlexGrid/Group` won't render properly (or as intended). I would suggest at the very least to just render the contents of `ActionCard` directly within `ActionCards` as to not separate the parent/child required relationship.